Template/BaseTemplate.php

```diff
@@ -27,9 +27,25 @@ abstract class BaseTemplate
 
     protected $templateType = '';
 
+    /**
+     * @deprecated
```

Comment:
can we add what to be used instead of this 
Eg: @deprecated Use FIELD_VARIABLE_COMPONENT instead